sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2105,6 +2105,22 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
     assert(error.getMessage contains "grouping_id() can only be used with GroupingSets/Cube/Rollup")
   }
 
+  test("uncorrelated scalar subquery") {
```

Comment:
About test coverage:

let's create a subquery suite and move the test cases there.

Also it would be great to have at least once test case that actually runs on a dataset that is not generated by just select x, because I worry in the future we add some special optimizations and then all the test cases here become no-op.